morph_tool/axon_point.py

```diff
@@ -0,0 +1,63 @@
+"""Module to detect the terminal point of the main axon/"""
+import logging
+import numpy as np
+
+from morphio import SectionType, IterType
+
+logger = logging.getLogger("morph_tool")
+
+
+def axon_point_section(morph, direction=None, bbox=None):
+    """Estimate axon point as the terminal point of the main axon.
+
+    This point is defined as the point for which the sum of the angles with the direction vector
+    is minimal. This corresponds to a main axon being a branch that follows best a straight line
+    in the given direction. More constraints can be added with bbox argument, if incorect axon is
+    detected.
+
+    Args:
+        morph (morphio.Morphology): morphology
+        direction (ndarray): estimated direction of main axon, if None [0, -1, 0] is used
+        bbox (dict): bbox of the form {'x': [-100, 100], 'y': [-500, -400]}
+
+    Returns:
+        MorphIO section ID for which the last point is the axon point
+    """
+    if direction is None:
+        direction = [0.0, -1.0, 0.0]
+    else:
+        direction /= np.linalg.norm(direction)
+
+    def _get_angle(section, direction):
+        """Get angle between section endpoints and direction."""
+        return np.arccos(
+            np.dot(direction, section.points[-1] - section.points[0])
+            / (np.linalg.norm(section.points[-1] - section.points[0]))
+        )
+
+    qualities, ids, positions = [], [], []
+    for section in morph.iter():
+        if section.type == SectionType.axon and not section.children:
+            qualities.append(
+                sum(_get_angle(section, direction) for section in section.iter(IterType.upstream))
+            )
+            ids.append(section.id)
+            positions.append(section.points[-1])
+
+    if bbox is not None:
+        qualities, positions, ids = np.array(qualities), np.array(positions), np.array(ids)
+        _convert = {"x": 0, "y": 1, "z": 2}
```

Comment:
In `morph_tool.apical_point.apical_point_position` we use `neurom.COLS`, I think it would be more consistent to also use it here.